Tests/ClangPlugin/test-runner.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024, Matthew Olsson <mattco@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/GenericLexer.h>
+#include <LibCore/Directory.h>
+#include <LibCore/Process.h>
+#include <LibCore/System.h>
+#include <LibDiff/Format.h>
+#include <LibDiff/Generator.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibFileSystem/TempFile.h>
+#include <LibTest/TestCase.h>
+
+static auto const path_to_compiler_binary = [] {
+    auto path_to_self = LexicalPath(MUST(Core::System::current_executable_path())).parent();
+    return LexicalPath::join(path_to_self.string(), "ClangPlugin"sv);
+}();
+
+static ByteString extract_expected_error(ByteBuffer const& buf)
+{
+    GenericLexer lexer { buf.bytes() };
+    ByteBuffer error;
+
+    while (lexer.consume_specific("/// "sv)) {
+        error.append(lexer.consume_until('\n').bytes());
+        error.append('\n');
+        lexer.ignore();
+    }
+
+    if (error.is_empty())
+        return {};
+
+    return ByteString { error.bytes() };
+}
+
+ErrorOr<ByteBuffer> read(ByteString const& path)
+{
+    auto file = TRY(Core::File::open(path, Core::File::OpenMode::Read));
+    return MUST(file->read_until_eof());
+}
+
+void find_tests(StringView path_to_search)
```

Comment:
Going to take a look at the CMake here in a sec, but my question about this test runner is: Would it be useful to use lit and FileCheck for these tests instead of our own custom test runner? If our analysis is generic enough that we would want to upstream it into `clang-analyzer` or such, it would be useful to be using the upstream llvm-project test runner.

Reply:
Not sure about lit and FileCheck, I can look into those, but I don't think anything here is really upstreamable. Maybe some checks in the future will be (since this is now a more generic tool), but atm I don't think so.

Reply:
Rewrote the tests with `lit`